packages/core/src/amazonqTest/error.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { ToolkitError } from '../shared/errors'
+
+export const technicalErrorCustomerFacingMessage =
+    'I am experiencing technical difficulties at the moment. Please try again in a few minutes.'
+const defaultTestGenErrorMessage = 'Amazon Q encountered an error while generating tests. Try again later.'
+export class TestGenError extends ToolkitError {
+    constructor(
+        error: string,
+        code: string,
+        public statusCode: string,
```

Comment:
can we make this a number

Reply:
Thought of using number but in our Telemetry statusCode is declared as String.
[aws-toolkit-common](https://github.com/aws/aws-toolkit-common/blob/5ebdd76f5dd424c35def8eef174fd837d5978819/telemetry/csharp/AwsToolkit.Telemetry.Events/Core/BaseTelemetryEvent.cs#L46-L50)

If required I can use number here and can convert to string while emitting?

Reply:
Addressed this as part of this [commit](https://github.com/aws/aws-toolkit-vscode/pull/6187/commits/f5e043e1c3a99e841a11a579fe94d06cc24807c2)